service/src/main/java/bio/terra/workspace/service/resource/controlled/flight/clone/workspace/ClonePolicyAttributesStep.java

```diff
@@ -41,11 +45,26 @@ public StepResult doStep(FlightContext context) throws InterruptedException, Ret
             .getInputParameters()
             .get(JobMapKeys.AUTH_USER_INFO.getKeyName(), AuthenticatedUserRequest.class);
 
-    Optional<ApiTpsPaoGetResult> workspacePao =
+    Optional<ApiTpsPaoGetResult> sourcePao =
         tpsApiDispatch.getPaoIfExists(
             new BearerToken(userRequest.getRequiredToken()), sourceWorkspaceId);
 
-    if (workspacePao.isPresent()) {
+    if (!sourcePao.isPresent()) {
+      // Source workspace doesn't have a PAO, so create one for it.
+      ApiTpsPaoCreateRequest request =
+          new ApiTpsPaoCreateRequest()
+              .objectId(sourceWorkspaceId)
+              .component(ApiTpsComponent.WSM)
+              .objectType(ApiTpsObjectType.WORKSPACE)
+              .attributes(new ApiTpsPolicyInputs());
+      tpsApiDispatch.createPao(new BearerToken(userRequest.getRequiredToken()), request);
```

Comment:
does this throw?

if so, return new StepResult(StepStatus.STEP_RESULT_FAILURE_RETRY, e);

same with deletePao. But perhaps that one throw FATAL instead?

Reply:
I can add that in. They don't throw anything in particular that I could look for, but they could throw something like "service unreachable". I'll have to catch the generic Exception.

Reply:
Create Pao can result in TPS `DuplicateObjectException` if WSM tries to create the sam Pao twice.